libvast/vast/detail/discard.hpp

```diff
@@ -14,9 +14,7 @@
 
 // Ensures all args are used and syntactically valid, without evaluating them.
 
-#if __GNUC__ == 8
-// GCC-8 does weird things here, so we actually evaluate the arguments (when
-// compiling without optimizations).
+#if __GNUC__
```

Comment:
Isn't this dependent on #1768?

Reply:
We switched to GCC-10 in our Dockerfiles recently and had lots of warnings in Release builds for that. This just enables the alternative discard macro for all versions of GCC instead of just GCC-8.